### PR TITLE
Document component approach

### LINKED ISSUE #137: Document Components in Readme
**Is your feature request related to a problem? Please describe.**
We currently don't explain what criteria are for our components to be added into Psammead. 

**Describe the solution you'd like**
Add a section in the Readme `Components`, explaining how we have only presentational components, and how Psammead is not a place for business logic - explain containers.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Documentation only. 

- [x] Initially labelled with ["Refinement needed"](https://github.com/BBC-News/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by bcmn**
Resolves #137 

The readme seems really approachable post-#174, so I was keen to keep it clean when approaching this ticket. As such, I figured we could link of to more in-depth explanations of the four intended sub-sections of Psammead (components, contexts, themes & utilities).

Based this on the notes coming out of the collaboration day the other week. In those, there's a lot of talk of examples or simple "dos & don'ts", but found those looked a little too much of an info-dump when I attempted to add some.

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,12 @@
 
 </div>
 
+Psammead is a [GEL-compliant](https://www.bbc.co.uk/gel/articles/what-is-gel) React component library for the BBC, built on [`styled-components`](https://www.styled-components.com). 
+
+ Psammead packages are split into:
+ * [Components](./packages/components/README.md) - GEL-compliant presentational React components, ready for use out of the box, regardless of data source.
+ * Utilities - Commonly shared Psammead dependencies, and fundamentals to aid building additional GEL-compliant components.
```
Comment (ChrisBAshton) [inline README.md]:
```suggestion
 * [Utilities](./packages/utilities) - Commonly shared Psammead dependencies, and fundamentals to aid building additional GEL-compliant components.
```

File: packages/components/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ Fundamentally, Psammead components are intended to be:
+ * Presentational
+ * GEL-Compliant
+ * Accessible
+
+ ## What does that mean?
+ ### Presentational
+ When developing and using Psammead components, we try to maintain a very clear distinction between [presentational and container components, as they are described in this blog post](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0). 
+
+ In summary, Psammead components should be almost exclusively concerned with how the component appears, and should avoid fetching or mutating data. Instead, these data fetching and mutating operations are split out into "container" components that can be entirely separate, and which can provide any necessary data or behaviour to presentational components via props.
```
Comment (ChrisBAshton) [inline packages/components/README.md]: Perhaps link to Simorgh as an example here.

File: packages/components/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Components
+
+ ## What is a Psammead component?
```
Comment (ChrisBAshton) [inline packages/components/README.md]: Dodgy markup in this file - everything after line 1 has a ` ` space before it 🤔

File: packages/components/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * GEL-Compliant
+ * Accessible
+
+ ## What does that mean?
+ ### Presentational
+ When developing and using Psammead components, we try to maintain a very clear distinction between [presentational and container components, as they are described in this blog post](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0). 
+
+ In summary, Psammead components should be almost exclusively concerned with how the component appears, and should avoid fetching or mutating data. Instead, these data fetching and mutating operations are split out into "container" components that can be entirely separate, and which can provide any necessary data or behaviour to presentational components via props.
+
+ These principles help Psammead components be reused across fundamentally different projects and back-ends.
```
Comment (ChrisBAshton) [inline packages/components/README.md]:
```suggestion
These principles help Psammead components to be reusable across different projects and back-ends.
```

**Review by ChrisBAshton — CHANGES_REQUESTED** [review]
Good start! Just fix the markdown 👍

**Review by ChrisBAshton — APPROVED** [review]
Approved. *signs*

**Review by sareh — APPROVED** [review]
:+1: Looks great!

**Pull request merged**